### PR TITLE
exchange: add SetRequester to exchange Base

### PR DESCRIPTION
**Pull request by geseq**
# PR Description

Hi again!

A small PR adding a helper function `setRequester` to `Base`.  

Currently, if I want to override the `Requester` I have to do it by updating the exported field from the `Base` struct, which doesn't lend itself very well to passing around an interface leading to duplicate code.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Test TestSetRequester

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes

### CODE REVIEW

File: exchanges/exchange_test.go [hunk 1]
```diff
@@ -2616,3 +2616,29 @@ func TestGetPairAndAssetTypeRequestFormatted(t *testing.T) {
 		t.Fatalf("received: '%v' but expected: '%v'", p, expected)
 	}
 }
+
+func TestSetRequester(t *testing.T) {
+	b := Base{
+		Config:    &config.Exchange{Name: "kitties"},
```
Comment (shazbert) [inline exchanges/exchange_test.go:2622]: 😹 👍

File: exchanges/exchange_test.go [hunk 2]
```diff
@@ -2616,3 +2616,29 @@ func TestGetPairAndAssetTypeRequestFormatted(t *testing.T) {
 		t.Fatalf("received: '%v' but expected: '%v'", p, expected)
 	}
 }
+
+func TestSetRequester(t *testing.T) {
+	b := Base{
```
Comment (shazbert) [inline exchanges/exchange_test.go:2621]: Can add `t.Parallel()` here.

File: exchanges/exchange.go [hunk 3]
```diff
@@ -47,6 +48,16 @@ var (
 	errSymbolCannotBeMatched             = errors.New("symbol cannot be matched")
 )
 
+// SetRequester sets the instance of the requester
+func (b *Base) SetRequester(r *request.Requester) error {
+	if r == nil {
+		return fmt.Errorf("%s cannot set requester, no requester provided", b.Name)
+	}
+
```
Comment (shazbert) [inline exchanges/exchange.go:56]: Suggestion: We could add a check here to see if its already been set?

**Review by shazbert — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good to me. Just a minor nit and a possible suggestion.

File: exchanges/exchange.go (hunk 3 above)
Comment (geseq) [reply exchanges/exchange.go:56]: I think there might be instances where it makes sense to replace this. For instance I like to just call the standard `SetupDefaults` that come with the exchange implementations which set a default Requester already, but want to replace it with my own.

File: exchanges/exchange_test.go (hunk 2 above)
Comment (geseq) [reply exchanges/exchange_test.go:2623]: Added

File: exchanges/exchange.go (hunk 3 above)
Comment (shazbert) [reply exchanges/exchange.go:56]: OK 👍

**Review by shazbert — APPROVED** [review]
Changes ACK!

**Review by thrasher- — APPROVED** [review]
Hello! and tACK 👍 Thanks for the PR

**Pull request merged**